test/testdata/rbi/array.rb

```diff
@@ -35,3 +35,9 @@
 # errors
 
 T::Array[Float].new.sum.nan? # error: Method `nan?` does not exist on `Integer` component of `T.any(Integer, Float)`
+
+# Sample
+T.assert_type!([1, 2].sample, Integer)
+T.assert_type!([1, 2].sample, random: Random.new, Integer)
+T.assert_type!([1, 2].sample(2), T::Array[Integer])
+T.assert_type!([1, 2].sample(2), random: Random.new, T::Array[Integer])
```

Comment:
Similarly, I guess you mean:
```suggestion
T.assert_type!([1, 2].sample(2, random: Random.new), T::Array[Integer])
```